src/parser/Mediatype.h

```diff
@@ -0,0 +1,263 @@
+//
+//  parser/Mediatype.h
+//  librefract
+//
+//  Created by Jiri Kratochvil on 02/08/2019
+//  Copyright (c) 2019 Apiary Inc. All rights reserved.
+//
+
+#ifndef PARSER_MEDIATYPE_H
+#define PARSER_MEDIATYPE_H
+
+#include "../PEGTL/include/tao/pegtl.hpp"
+
+#include <string>
+#include <map>
+
+#include <iostream>
+
+namespace parser 
+{
+
+    namespace mediatype {
+
+        namespace pegtl = tao::pegtl;
+
+        struct restricted : pegtl::one< '!', '#', '$', '&', '^',  '_', '-', '.' >  {};
```

Comment:
I used this one set https://tools.ietf.org/html/rfc6838#section-4.2
Also name `restricted` comes from here

Reply:
Ok, it seems media-type is defined across two RFCs, RFC6838 which defines restricted which applies to type/subtype etc, this RFC does not declare the requirements for parameters, that seems to be in RFC2616 which uses the separators listed here.

Reply:
It define requirements for attributes here
https://tools.ietf.org/html/rfc6838#section-4.3

```
parameter-name = restricted-name

Note that this syntax is somewhat more restrictive than what is
allowed by the ABNF in [RFC2045] and amended by [RFC2231].
```

Reply:
But also:

> There is no defined syntax for parameter values.  Therefore, registrations MUST specify parameter value syntax.

So it seems it will be better use RFC2616 for parameter value parsing
(Or for full implementation?)